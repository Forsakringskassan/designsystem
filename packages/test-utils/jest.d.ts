export * from "./dist/types/jest";

declare global {
    namespace jest {
        interface Matchers<R> {
            /**
             * Expects element to have focus.
             */
            toHaveFocus(): R;
        }
    }
}
