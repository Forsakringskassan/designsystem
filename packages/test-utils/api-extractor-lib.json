{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/lib.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/lib.d.ts"
    }
}
