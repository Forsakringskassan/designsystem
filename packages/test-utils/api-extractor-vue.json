{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "mainEntryPointFilePath": "<projectFolder>/temp/types/vue/index.d.ts",
    "apiReport": {
        "enabled": false
    },
    "dtsRollup": {
        "publicTrimmedFilePath": "<projectFolder>/dist/types/vue.d.ts"
    }
}
