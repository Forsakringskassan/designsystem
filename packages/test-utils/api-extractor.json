{
    "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
    "extends": "../../api-extractor.json",
    "tsdocMetadata": {
        "tsdocMetadataFilePath": "<projectFolder>/dist/types/tsdoc-metadata.json"
    },
    "dtsRollup": {
        "enabled": false
    }
}
