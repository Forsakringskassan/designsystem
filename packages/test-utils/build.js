const fs = require("node:fs/promises");
const path = require("node:path");
const esbuild = require("esbuild");

const type = {
    cjs: "commonjs",
    esm: "module",
};

async function build() {
    for (const format of ["cjs", "esm"]) {
        const outdir = `dist/${format}`;
        const common = {
            logLevel: "info",
            sourcemap: true,
            external: ["vue", "@fkui/*"],
            bundle: true,
            platform: "node",
            format,
            target: "node20",
            outdir,
        };
        await esbuild.build({
            ...common,
            entryPoints: ["src/lib.ts", "src/jest.ts"],
        });
        await esbuild.build({
            ...common,
            entryPoints: [{ in: "src/vue/index.ts", out: "vue" }],
            define: {
                "process.env.NODE_ENV": JSON.stringify("development"),
            },
        });
        const pkg = JSON.stringify({ type: type[format] }, null, 2);
        await fs.writeFile(path.join(outdir, "package.json"), pkg, "utf-8");
    }
}

build().catch((err) => {
    console.error(err);
});
