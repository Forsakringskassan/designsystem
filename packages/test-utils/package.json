{
  "name": "@fkui/test-utils",
  "version": "6.6.0",
  "description": "FKUI test utils for Jest",
  "keywords": [
    "fkui",
    "designsystem",
    "test",
    "jest",
    "vue",
    "cypress"
  ],
  "homepage": "https://designsystem.forsakringskassan.se/",
  "bugs": "https://github.com/Forsakringskassan/designsystem/issues",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Forsakringskassan/designsystem.git",
    "directory": "packages/test-utils"
  },
  "license": "MIT",
  "author": "Försäkringskassan",
  "sideEffects": [
    "./dist/cjs/jest.js",
    "./dist/esm/jest.js",
    "./src/matchers/augmentations.ts",
    "./src/matchers/index.ts"
  ],
  "exports": {
    ".": {
      "types": "./dist/types/lib.d.ts",
      "import": "./dist/esm/lib.js",
      "require": "./dist/cjs/lib.js"
    },
    "./jest": {
      "types": "./dist/types/jest.d.ts",
      "import": "./dist/esm/jest.js",
      "require": "./dist/cjs/jest.js"
    },
    "./vue": {
      "types": "./dist/types/vue.d.ts",
      "import": "./dist/esm/vue.js",
      "require": "./dist/cjs/vue.js"
    }
  },
  "main": "dist/cjs/lib.js",
  "types": "dist/types/lib.d.ts",
  "files": [
    "dist",
    "jest.d.ts",
    "vue.d.ts"
  ],
  "scripts": {
    "prebuild": "rimraf dist temp && tsc",
    "build": "node build.js",
    "postbuild": "fk-api-extractor --patch-augmentations api-extractor.json api-extractor-*.json",
    "clean": "rimraf .jest-cache dist temp test-results",
    "prepack": "release-prepack --bundle --retain-scripts",
    "postpack": "release-postpack",
    "prepublishOnly": "release-prepublish --bundle --retain-scripts",
    "test": "jest"
  },
  "jest": {
    "preset": "@forsakringskassan/jest-config-vue"
  },
  "devDependencies": {
    "@vue/server-renderer": "3.5.13",
    "@vue/test-utils": "2.4.6",
    "esbuild": "0.25.3",
    "jest-environment-jsdom": "29.7.0",
    "vue": "3.5.13"
  },
  "peerDependencies": {
    "jest": "^26.0.0 || ^27.0.0 || ^28.0.0 || ^29.0.0",
    "vue": "^3.0.0"
  },
  "peerDependenciesMeta": {
    "vue": {
      "optional": true
    }
  },
  "engines": {
    "node": ">= 20",
    "npm": ">= 7"
  }
}
