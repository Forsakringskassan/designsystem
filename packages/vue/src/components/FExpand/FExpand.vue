<script lang="ts">
import { defineComponent } from "vue";
import { getHTMLElementFromVueRef } from "../../utils";

export default defineComponent({
    name: "FExpand",
    data() {
        return {
            height: 0,
            initialStyle: {
                overflow: "hidden",
                transition: "height 400ms cubic-bezier(0.46, 0.03, 0.52, 0.96)",
            },
            hiddenStyle: {
                height: "auto",
                position: "absolute",
                visibility: "hidden",
            },
            visibleStyle: {
                width: "",
                position: "",
                visibility: "",
                height: "0px",
            },
            openedStyle: {
                height: "auto",
            },
        };
    },
    methods: {
        enter(element: Element): void {
            const htmlElement = getHTMLElementFromVueRef(element);
            Object.assign(htmlElement.style, this.initialStyle);
            Object.assign(htmlElement.style, this.hiddenStyle);
            htmlElement.style.width = getComputedStyle(element).width;
            const height = getComputedStyle(element).height;
            Object.assign(htmlElement.style, this.visibleStyle);
            // Force redraw
            /* eslint-disable-next-line @typescript-eslint/no-unused-expressions -- technical debt, there should be a better way */
            getComputedStyle(element).height;
            setTimeout(() => {
                this.height = parseInt(height as string, 10);
                htmlElement.style.height = height;
            });
        },
        afterEnter(element: Element): void {
            const htmlElement = getHTMLElementFromVueRef(element);
            Object.assign(htmlElement.style, this.openedStyle);
        },
        leave(element: Element): void {
            const htmlElement = getHTMLElementFromVueRef(element);
            const height = getComputedStyle(element).height;
            htmlElement.style.height = height;

            // Force redraw
            /* eslint-disable-next-line @typescript-eslint/no-unused-expressions -- technical debt, there should be a better way */
            getComputedStyle(element).height;
            setTimeout(() => {
                Object.assign(htmlElement.style, this.visibleStyle);
            });
        },
    },
});
</script>

<template>
    <transition @enter="enter" @after-enter="afterEnter" @leave="leave">
        <!-- @slot Content to expand -->
        <slot :height="height"></slot>
    </transition>
</template>
