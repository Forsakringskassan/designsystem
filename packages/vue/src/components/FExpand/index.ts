export { default as FExpand } from "./FExpand.vue";
