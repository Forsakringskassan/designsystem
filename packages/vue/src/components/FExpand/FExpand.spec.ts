import { VueWrapper, mount } from "@vue/test-utils";
import FExpand from "./FExpand.vue";

function createWrapper({
    props = {},
    slots = {},
    attrs = {},
} = {}): VueWrapper {
    return mount(FExpand, {
        attrs: { ...attrs },
        props: { ...props },
        slots: { ...slots },
    });
}

describe("snapshots", () => {
    it("should match snapshot with default slot content", () => {
        const wrapper = createWrapper({
            slots: {
                default: /* HTML */ ` <p>Content</p> `,
            },
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
