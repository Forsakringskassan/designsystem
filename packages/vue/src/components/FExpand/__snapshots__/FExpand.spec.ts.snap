// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should match snapshot with default slot content 1`] = `
<transition-stub
  appear="false"
  css="true"
  persisted="false"
>
  
  <p>
    Content
  </p>
  
</transition-stub>
`;
